spring-mvc-simple-2/src/main/java/com/baeldung/exclude_urls_filter/filter/HeaderContextFilter.java

```diff
@@ -0,0 +1,27 @@
+package com.baeldung.exclude_urls_filter.filter;
+
+import com.baeldung.exclude_urls_filter.utils.HeaderContextProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+@Order(1)
+public class HeaderContextFilter implements Filter {
+
+	@Override
+    public void doFilter(
+            ServletRequest request,
+            ServletResponse response,
+            FilterChain chain) throws IOException, ServletException {
+
+        String countryCodeHeader = ((HttpServletRequest) request).getHeader("COUNTRY_CODE");
+        HeaderContextProvider.setContext(countryCodeHeader);
+        chain.doFilter(request, response);
```

Comment:
I am not totally sure but shouldn't that be protected in such a way:
HeaderContextProvider.setContext(countryCodeHeader);
  try {
    chain.doFilter(request, response);
} finally {
    HeaderContextProvider.unsetContext();
}

Reply:
Done 🙂